src/publishMain.ml

```diff
@@ -595,8 +600,88 @@ let to_files ?(split=false) ~packages_dir meta_opams =
     []
   |> List.rev
 
-let main_term root =
-  let run args force tag version dry_run repo target_branch packages_dir title msg split =
+let check_depends_stanza meta_opams opam_root =
+  OpamConsole.note "Checking 'depends' stanza in your opam files \
+                    (bypass with --skip-deps-check)";
+  let config =
+    match OpamStateConfig.load_defaults opam_root with
+    | Some config -> config
+    | None -> failwith "OpamStateConfig.load_defaults"
+  in
+  let switch =
+    match OpamFile.Config.switch config with
+    | Some switch -> switch
+    | None -> failwith "OpamFile.Config.switch"
+  in
+  OpamGlobalState.with_ `Lock_none @@ fun gt ->
+  OpamSwitchState.with_ `Lock_none gt ~switch @@ fun st ->
+  let direct_deps =
+    OpamPackage.Map.fold (fun pkg (_meta, opam_file) acc ->
+      let depends = OpamFile.OPAM.depends opam_file in
+      let env = OpamPackageVar.resolve_switch ~package:pkg st in
+      let formula = OpamFilter.filter_formula env depends in
+      OpamFormula.ors [acc; formula])
+      meta_opams OpamFormula.Empty
+  in
+  let resolved_deps =
+    let direct_pkgs =
+      OpamFormula.packages_of_atoms
+        (OpamPackage.Set.union st.installed st.packages)
+        (OpamFormula.atoms direct_deps)
+    in
+    let transitive_pkgs =
+      let universe =
+        OpamSwitchState.universe st
+          ~requested:(OpamPackage.names_of_packages st.packages)
+          Query
+      in
+      OpamSolver.dependencies ~depopts:false ~build:false ~post:false
+        ~installed:false ~unavailable:false
+        universe direct_pkgs
+      |> OpamPackage.Set.of_list
+    in
+    OpamPackage.Set.union direct_pkgs transitive_pkgs
+  in
+  let missing_direct_deps =
+    OpamFormula.fold_left (fun acc (name, _) ->
+      if not (OpamPackage.Set.exists (fun pkg ->
+        OpamPackage.name pkg = name) resolved_deps)
+      then OpamPackage.Name.to_string name :: acc
+      else acc)
+      [] direct_deps
```

Comment:
I need to see more in details this part, I thing that we can do it simpler and without solver.